### PR TITLE
[extractor/twitter] Fix `--no-playlist` and add media `view_count` when using GraphQL

### LINKED ISSUE #6210: [twitter] --no-playlist param leads to failed download for posts with image+video
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2023.01.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

Not sure, if it's a bug or not, but still decided to report this one:

Test link:
https://twitter.com/hlo_again/status/1599108751385972737/video/2
This post has an image and video in it. download works just fine and video is found properly.
However, if --no-playlist param is used it fails with ERROR: [twitter] 1599108751385972737: Video #2 is unavailable
Log below. 

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
./yt-dlp -vU --no-playlist -F "https://twitter.com/hlo_again/status/1599108751385972737/video/2"
[debug] Command-line config: ['-vU', '--no-playlist', '-F', 'https://twitter.com/hlo_again/status/1599108751385972737/video/2']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.01.06 [6becd2508] (zip)
[debug] Python 3.10.10 (CPython arm64 64bit) - macOS-12.6.3-arm64-arm-64bit (OpenSSL 1.1.1t  7 Feb 2023)
[debug] exe versions: phantomjs 2.1.1
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Extractor Plugins: SamplePluginIE
[debug] Post-Processor Plugins: SamplePluginPP
[debug] Loaded 1824 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.01.06, Current version: 2023.01.06
yt-dlp is up to date (2023.01.06)
[twitter] Extracting URL: https://twitter.com/hlo_again/status/1599108751385972737/video/2
[twitter] 1599108751385972737: Downloading guest token
[twitter] 1599108751385972737: Downloading JSON metadata
[debug] [twitter] Extracting from video info: 1599108643743473680
[twitter] 1599108751385972737: Downloading m3u8 information
[twitter] Downloading just the URL-specified video number 2 because of --no-playlist
ERROR: [twitter] 1599108751385972737: Video #2 is unavailable
  File "./yt-dlp/yt_dlp/extractor/common.py", line 693, in extract
    ie_result = self._real_extract(url)
  File "./yt-dlp/yt_dlp/extractor/twitter.py", line 1072, in _real_extract
    raise ExtractorError(f'Video #{selected_index} is unavailable', expected=True)
```

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixes the behavior of `--no-playlist` when image media is present. It also now adds the old archive id to the post and adds `#2` to the title like with playlist items.

It also adds extraction of `view_count` when using GraphQL (per media item, not per tweet). This makes all native video tests fail when using GraphQL since `view_count` is not defined in the test. A specific test for using `force_graphql` extractor arg has been added instead.

Fixes #6210
Addresses one part of #5807

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/twitter.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ class TwitterIE(TwitterBaseIE):
             'id': '665052190608723968',
             'display_id': '665052190608723968',
             'ext': 'mp4',
-            'title': 'md5:e99588f17b3dd0503814ffb560e64731',
+            'title': str,
```
Comment (pukkandan) [inline yt_dlp/extractor/twitter.py:339]: ?

File: yt_dlp/extractor/twitter.py [hunk 2]
```diff
@@ -1052,11 +1092,31 @@ def get_binding_value(k):
                         'content_duration_seconds')),
                 }
 
-        media_path = ((None, 'quoted_status'), 'extended_entities', 'media', lambda _, m: m['type'] != 'photo')
-        videos = map(extract_from_video_info, traverse_obj(status, media_path, expected_type=dict))
-        cards = extract_from_card_info(status.get('card'))
-        entries = [{**info, **data, 'display_id': twid} for data in (*videos, *cards)]
+        videos = traverse_obj(status, ((
+            None, 'quoted_status'), 'extended_entities', 'media', lambda _, m: m['type'] != 'photo', {dict}))
```
Comment (pukkandan) [inline yt_dlp/extractor/twitter.py:1096]:
```suggestion
        videos = traverse_obj(status, (
            (None, 'quoted_status'), 'extended_entities', 'media', lambda _, m: m['type'] != 'photo', {dict}))
```

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/twitter.py (hunk 1 above)
Comment (Grub4K) [reply yt_dlp/extractor/twitter.py:339]: `title` is composed of `{author} - {description}` where on this account the `author` constantly changes, hence why I was putting it as `str`. Alternatively these are also sensible:
- Use the full description as string instead so the test fail diff shows that only the author name changed
- Regex match only the start , copying the `author` regex
- Just skip the test bc its unreliable.

Comment (pukkandan) [reply yt_dlp/extractor/twitter.py:339]: imo regex is the best option here. use some of the actual description and let the rest be .* so we know we are extracting the right field

**Pull request merged**